### PR TITLE
🪟  🐛  Split Heading and Text components

### PR DESCRIPTION
**Pull request by joeykmh**
## What
- Fixes a runtime warning: `Warning: Received 'true' for a non-boolean attribute 'bold'.`, which appears whenever `<Text bold />` is used in development mode.

This arises because `bold: boolean` is inadvertently spread onto the HTML element via `...remainingProps` in [Text.tsx](https://github.com/airbytehq/airbyte/blob/master/airbyte-webapp/src/components/ui/Text/Text.tsx#L55):

```jsx
  // Text.tsx

  return React.createElement(as, {
    ...remainingProps,
    className,
    children,
  });
```

## How
The underlying problem is that `<Text>` supports two component signatures (`TextProps | HeadingProps`) _and_ passes through any additional props. This makes it hard to intercept the props we care about while passing the rest.

IMO it's simpler to be declarative rather than imperative here, and replace one complex component with two simple ones.

This PR:
- Splits `<Text>` into `<Heading>` and `<Text>` so we don't have to support two prop signatures
- Updates the codebase to use `<Heading>` where appropriate
- Adds `<Heading>` to Storybook

@edmundito would be great to get your review here because I see you [decided against this approach](https://github.com/airbytehq/airbyte/pull/15570#discussion_r944511106) previously.

## Recommended reading order
1. `Text.tsx` & `Heading.tsx`
2. The rest

### CODE REVIEW

**Review by edmundito — APPROVED** [review]
I'm ok with this change. I originally proposed having a single component for maintenance but the implementation didn't turn out as simple as expected and this approach feels easier to maintain. I also like how in the usage heading and text read separately.

File: airbyte-webapp/src/components/ui/PageHeader/PageHeader.tsx [hunk 1]
```diff
@@ -21,25 +21,25 @@ export const PageHeader: React.FC<PageHeaderProps> = ({
   endComponent,
 }) => (
   <div className={classNames(styles.container)} data-withline={withLine}>
-    <Text
+    <Heading
       as="h1"
```
Comment (dizel852) [inline airbyte-webapp/src/components/ui/PageHeader/PageHeader.tsx:25]:
```suggestion
```
Super nit: don't need it anymore 😊

File: airbyte-webapp/src/packages/cloud/views/auth/SignupPage/SignupPage.tsx [hunk 2]
```diff
@@ -20,7 +20,7 @@ const SignupPage: React.FC<SignupPageProps> = ({ highlightStyle }) => {
   return (
     <div>
       <HeadTitle titles={[{ id: "login.signup" }]} />
-      <Text as="h1" size="xl" className={styles.title}>
+      <Heading as="h1" size="xl" className={styles.title}>
```
Comment (dizel852) [inline airbyte-webapp/src/packages/cloud/views/auth/SignupPage/SignupPage.tsx:23]:
```suggestion
      <Heading size="xl" className={styles.title}>
```

File: airbyte-webapp/src/packages/cloud/views/auth/components/FormTitle/FormTitle.tsx [hunk 3]
```diff
@@ -1,11 +1,11 @@
 import React from "react";
 
-import { Text } from "components/ui/Text";
+import { Heading } from "components/ui/Heading";
 
 import styles from "./FormTitle.module.scss";
 
 export const FormTitle: React.FC<React.PropsWithChildren<unknown>> = ({ children }) => (
-  <Text as="h1" size="xl" className={styles.title}>
+  <Heading as="h1" size="xl" className={styles.title}>
```
Comment (dizel852) [inline airbyte-webapp/src/packages/cloud/views/auth/components/FormTitle/FormTitle.tsx:8]:
```suggestion
  <Heading size="xl" className={styles.title}>
```

File: airbyte-webapp/src/packages/cloud/views/workspaces/WorkspacesPage/WorkspacesPage.tsx [hunk 4]
```diff
@@ -14,10 +15,10 @@ const WorkspacesPage: React.FC = () => {
   return (
     <div className={styles.container}>
       <img className={styles.logo} alt="logo" src="/cloud-main-logo.svg" width={186} />
-      <Text as="h1" size="lg" centered>
+      <Heading as="h1" size="lg" centered>
```
Comment (dizel852) [inline airbyte-webapp/src/packages/cloud/views/workspaces/WorkspacesPage/WorkspacesPage.tsx:18]:
```suggestion
      <Heading size="lg" centered>
```

**Review by dizel852 — APPROVED** [review]
Code LGTM 👍  Also tested locally
Left some super nit suggestions:
Since we have default `as` and `size` props, we don't need to overwrite them:
`h1` (for `<Heading />)` and `p` (for `<Text />`) as `as` and `md` as `size`.

**Comment by josephkmh** [comment]
> Since we have default as and size props, we don't need to overwrite them:
> `h1` (for `<Heading />`) and `p` (for `<Text />`) as `as` and `md` as s`ize`.

@dizel852 good point! On second thought actually I think a default of `h1` might be a little aggressive... It might be better to force consumers to specify which level heading they intend, since it has content hierarchy/accessibility implications. Otherwise I can imagine someone using `<Heading>Some sub-heading</Heading>` and not realizing they're putting an `<h1>` on the page.  WDYT?

A default of `p` for `<Text>` is a little less dangerous, so I'd be inclined to leave that one. And the default sizes we can definitely remove 🙂

**Comment by dizel852** [comment]
> On second thought actually I think a default of `h1` might be a little aggressive...

Oh yeah, duly noted.

> I can imagine someone using `<Heading>Some sub-heading</Heading>` and not realizing they're putting an `<h1>` on the page.

Agree


> A default of `p` for `<Text>` is a little less dangerous, so I'd be inclined to leave that one. And the default sizes we can definitely remove 🙂

👍 

So, will we remove the `h1` as a default `as` prop for `<Heading>`? To make someone explicitly write what kind of heading he'd like to use.

@edmundito probably you have some thoughts?

**Comment by josephkmh** [comment]
> So, will we remove the h1 as a default as prop for <Heading>? To make someone explicitly write what kind of heading he'd like to use.

Yes exactly, I've removed the default value now and it's a required prop.

File: airbyte-webapp/src/components/base/Titles/Titles.tsx [hunk 5]
```diff
@@ -19,13 +19,13 @@ const H1 = styled.h1<IProps>`
   margin: 0;
 `;
 
-/** @deprecated Use `<Text as="h1 | h2 | h3 | h4 | h5" size="md" />` */
+/** @deprecated Use `<Heading as="h3" />` */
```
Comment (edmundito) [inline airbyte-webapp/src/components/base/Titles/Titles.tsx:22]: The comments were ok before, except for the `Text` component name change. The h3 is being misused in this implementation so when replacing a new they'd have to pick the most appropriate h type, but the size for this is equivalent to "md"

**Review by edmundito — APPROVED** [review]
Re-reviewed and re-approved!

File: airbyte-webapp/src/components/base/Titles/Titles.tsx (hunk 5 above)
Comment (josephkmh) [reply airbyte-webapp/src/components/base/Titles/Titles.tsx:22]: Got it, will revert the comment except for the component name change!

**Pull request merged**